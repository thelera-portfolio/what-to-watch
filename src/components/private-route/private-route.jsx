import {AppRoute} from "../../utils/consts.js";
import {AuthorizationStatus} from "../../reducer/user/user.js";
import {connect} from "react-redux";
import {getAuthorizationStatus} from "../../reducer/user/selectors.js";
import PropTypes from "prop-types";
import React from "react";
import {Redirect, Route} from "react-router-dom";

const PrivateRoute = (props) => {
  const {authorizationStatus, exact, path, render} = props;

  return (
    <Route
      path={path}
      exact={exact}
      render={(properties) => {
        return (
          authorizationStatus === AuthorizationStatus.AUTH
            ? render(properties)
            : <Redirect to={AppRoute.LOGIN} />
        );
      }}
    />
  );
};

PrivateRoute.propTypes = {
  authorizationStatus: PropTypes.string.isRequired,
  exact: PropTypes.bool.isRequired,
  path: PropTypes.string.isRequired,
  render: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  authorizationStatus: getAuthorizationStatus(state),
});

export {PrivateRoute};
export default connect(mapStateToProps)(PrivateRoute);
