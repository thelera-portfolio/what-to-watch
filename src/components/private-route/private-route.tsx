import * as React from "react";
import {AppRoute} from "../../utils/consts";
import {AuthorizationStatus} from "../../reducer/user/user";
import {connect} from "react-redux";
import {getAuthorizationStatus} from "../../reducer/user/selectors";
import {Redirect, Route} from "react-router-dom";

const PrivateRoute = (props) => {
  const {authorizationStatus, exact, path, render} = props;

  return (
    <Route
      path={path}
      exact={exact}
      render={(properties) => {
        return (
          authorizationStatus === AuthorizationStatus.AUTH
            ? render(properties)
            : <Redirect to={AppRoute.LOGIN} />
        );
      }}
    />
  );
};

PrivateRoute.propTypes = {
  authorizationStatus: PropTypes.string.isRequired,
  exact: PropTypes.bool.isRequired,
  path: PropTypes.string.isRequired,
  render: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  authorizationStatus: getAuthorizationStatus(state),
});

export {PrivateRoute};
export default connect(mapStateToProps)(PrivateRoute);
