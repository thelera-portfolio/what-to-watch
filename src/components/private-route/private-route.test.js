import {BrowserRouter} from "react-router-dom";
import {PrivateRoute} from "./private-route.jsx";
import React from "react";
import renderer from "react-test-renderer";

it(`PrivateRoute is rendered correctly`, () => {
  const tree = renderer.create(
      <BrowserRouter>
        <PrivateRoute
          authorizationStatus={`AUTH`}
          exact={true}
          path={`path`}
          render={() => {}}
        />
      </BrowserRouter>).toJSON();

  expect(tree).toMatchSnapshot();
});
