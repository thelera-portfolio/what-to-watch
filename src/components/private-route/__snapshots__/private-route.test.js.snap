// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute is rendered correctly 1`] = `null`;
